src/reference/asciidoc/mqtt.adoc

```diff
@@ -421,4 +424,74 @@ MqttConnectOptions options = source.getConnectionInfo();
+It requires a `clientId` and MQTT broker URL or `MqttConnectionOptions` reference.
+It does support a `MqttClientPersistence` option, can be `async` and can emit `MqttIntegrationEvent` objects in that case (see `asyncEvents` option).
+If request message payload is an `org.eclipse.paho.mqttv5.common.MqttMessage`, it is published as is via internal `IMqttAsyncClient`.
+If the payload is `byte[]` it is used as is for target `MqttMessage` payload to publish.
+If the payload is a `String` it is converted to `byte[]` to publish.
+The rest use-cases are delegated to the provided `MessageConverter` which is a `IntegrationContextUtils.ARGUMENT_RESOLVER_MESSAGE_CONVERTER_BEAN_NAME` `ConfigurableCompositeMessageConverter` bean from the application context.
+Note: the provided `HeaderMapper<MqttProperties>` is not used when the requested message payload is already an `MqttMessage`.
+The following Java DSL configuration sample demonstrates how to use this channel adapter in the integration flow:
+
+====
+[source, java]
+----
+@Bean
+public IntegrationFlow mqttOutFlow() {
+    Mqttv5PahoMessageHandler messageHandler = new Mqttv5PahoMessageHandler(MQTT_URL, "mqttv5SIout");
+    MqttHeaderMapper mqttHeaderMapper = new MqttHeaderMapper();
+    mqttHeaderMapper.setOutboundHeaderNames("some_user_header", MessageHeaders.CONTENT_TYPE);
+    messageHandler.setHeaderMapper(mqttHeaderMapper);
+    messageHandler.setAsync(true);
+    messageHandler.setAsyncEvents(true);
+    messageHandler.setConverter(mqttStringToBytesConverter());
+
+    return f -> f.handle(messageHandler);
+}
+----
+====
+
+IMPORTANT: The `org.springframework.integration.mqtt.support.MqttMessageConverter` cannot be used with the `Mqttv5PahoMessageHandler` since its contract is aimed only for MQTT v3 protocol.
+
+See more information in the `Mqttv5PahoMessageHandler` javadocs and its superclass.
+
+The inbound channel adapter for MQTT v5 protocol is present as an `Mqttv5PahoMessageDrivenChannelAdapter`.
+It requires a `clientId` and MQTT broker URL or `MqttConnectionOptions` reference, plus topics to subscribe and consume.
+It does support a `MqttClientPersistence` option, which is in-memory by default.
+The expected `payloadType` (`byte[]` by default) can be configured and it is propagated to the provided `SmartMessageConverter` for conversion from `byte[]` of the received `MqttMessage`.
+If the `manualAck` option is set, then an `IntegrationMessageHeaderAccessor.ACKNOWLEDGMENT_CALLBACK` header is added to the message to produce as an instance of `SimpleAcknowledgment`.
+The `HeaderMapper<MqttProperties>` is used to map `PUBLISH` frame properties (including user properties) into the target message headers.
+Standard `MqttMessage` properties, such as `qos`, `id`, `dup`, `retained`, plus received topic are always mapped to headers.
+See `MqttHeaders` for more information.
+
+The following Java DSL configuration sample demonstrates how to use this channel adapter in the integration flow:
```

Comment:
```suggestion
IMPORTANT: The `org.springframework.integration.mqtt.support.MqttMessageConverter` cannot be used with the `Mqttv5PahoMessageHandler` since its contract is aimed only for the MQTT v3 protocol.

See more information in the `Mqttv5PahoMessageHandler` javadocs and its superclass.

The inbound channel adapter for the MQTT v5 protocol is present as an `Mqttv5PahoMessageDrivenChannelAdapter`.
It requires a `clientId` and MQTT broker URL or `MqttConnectionOptions` reference, plus topics to which to subscribe and consume from.
It supports a `MqttClientPersistence` option, which is in-memory by default.
The expected `payloadType` (`byte[]` by default) can be configured and it is propagated to the provided `SmartMessageConverter` for conversion from `byte[]` of the received `MqttMessage`.
If the `manualAck` option is set, then an `IntegrationMessageHeaderAccessor.ACKNOWLEDGMENT_CALLBACK` header is added to the message to produce as an instance of `SimpleAcknowledgment`.
The `HeaderMapper<MqttProperties>` is used to map `PUBLISH` frame properties (including user properties) into the target message headers.
Standard `MqttMessage` properties, such as `qos`, `id`, `dup`, `retained`, plus received topic are always mapped to headers.
See `MqttHeaders` for more information.

The following Java DSL configuration sample demonstrates how to use this channel adapter in the integration flow:
```